File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.py
def solution(begin, end):
    answer = []
    
    for x in range(begin, end + 1):
        if x == 1:
            answer.append(0)
            continue

        temp = 1
        for i in range(2, int(x**0.5) + 1):
            if x % i == 0:
                if x // i <= 10000000:
                    temp = x // i
                    break
                else:
                    temp = i

        answer.append(temp)
        
    return answer

File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.js
function solution(begin, end) {
    const result = [];

    for (let i = begin; i <= end; i++) {
        if (i === 1) {
            result.push(0);
            continue;
        }
        
        result.push(getMaxDivisor(i));
    }

    return result;
}

function getMaxDivisor(num) {
    let temp = 1;
    for (let i = 2; i <= Math.sqrt(num); i++) {
        if (num % i === 0) {
            if (num / i <= 10000000) {
                return num / i;
            }            
            temp = i;
        }
    }
    return temp;
}